### DISCUSSION TITLE
How important is dark mode?

### DISCUSSION THREAD

**Discussion by roblevintennis** [AgnosticUI/agnosticui#130] [Q&A]
If you look at the issue tracker you can see that I've prioritized gaining a certain amount of component coverage as v1 MVP and dark mode is labelled `v2`. I suspect this is a nuanced and debatable topic.

I'd love to hear folks chime in if they strongly disagree or have any feedback there. I doubt it's going to be "that hard" to support dark mode but I'm just trying to prioritize things given I have a limited amount of time to work on AgnosticUI weekly. Thought y'all?

**Comment by mgan59** [comment]
I place more value into having a functional set of component/coverage.  But I do understand the folks out there that place `dark-mode` support has a _must-have_.  I do think it is helpful to have some level of a `theme-provider` that you see in a front-end library like `styled-components`

**Reply by roblevintennis** [reply]
Since AgnosticUI is not a css in js framework (Vue and react implementations use css modules, svelte component level obfuscation, and Angular view encapsulation), I'm not sure we could do a theme object literal approach.

Currently the approach is via css custom properties and I think I want to stick with that in terms of the technical means to theme.

However, I DO think we could evolve and improve there. I'm not yet happy with the variables themselves. They're a bit inconsistent. But it's partially the way I work and iterate sort of discovering what's working best.

That said, I'll plan to likely revisit these variables and see what can be improved. Thanks for sharing your feedback and input!

**Comment by roblevintennis** [comment]
I'm closing out legacy discussions as the v2 rewrite is almost ready and is a complete rewrite overhaul so discussions are no longer relevant at this point.
